source/docs/software/advanced-gradlerio/deploy-git-data.rst

```diff
@@ -0,0 +1,100 @@
+Including Git Data in Deploy
+============================
+
+This article will go over how to include information from Git, such as branch name or commit hash, into the robot code. This is necessary for using such information in robot code, such as printing out commit hash and branch name when the robot starts.
+
+.. note:: Git must be in the path for this to work. This should be enabled by default when `installing Git <https://git-scm.com/downloads>`__.
+
+Deploying Branch Name
+---------------------
+
+This example uses `git rev-parse <https://git-scm.com/docs/git-rev-parse>`__ to extract data the name of the current branch. The Git command used for this is:
+
+.. code-block:: console
+
+   $ git rev-parse --abbrev-ref HEAD
+
+The ``--abbrev-ref`` flag tells Git to use a short version of the name for the current commit that ``rev-parse`` is acting on. When HEAD is the most recent commit on a branch, this will return the name of that branch.
+
+Next, create a new task in the ``build.gradle`` file that will run the above Git command and write it to a file in the ``src/main/deploy`` directory. For example, the following is an example task named ``writeBranchName`` that will write the branch name to a file named ``branch.txt``.
+
+.. code-block:: groovy
+
+   tasks.register("writeBranchName") {
+      // Define an output stream to write to instead of terminal
+      def stdout = new ByteArrayOutputStream()
+
+      // Execute the git command
+      exec {
+         commandLine "git", "rev-parse", "--abbrev-ref", "HEAD"
+         // Write to the output stream instead of terminal
+         standardOutput = stdout
+      }
+
+      // Parse the output into a string
+      def branch = stdout.toString().trim()
+
+      // Create a new file
+      new File(
+         // Join project directory and deploy directory
+         projectDir.toString() + "/src/main/deploy",
+         // File name to write to
+         "branch.txt"
+      ).text = branch // Set the contents of the file to the variable branch
+   }
```

Comment:
@ThadHouse does this work for 2022 gradlerio?

Reply:
yes

Reply:
Just tested it with a project I imported into 2022 and it seems to work just fine

Reply:
See comment below. Its wrong for both 2021 and 2022. I posted the fix for 2022.